File: ios/RoutineChart/RoutineChart/Core/Domain/Repositories/RoutineRepository.swift
//
//  RoutineRepository.swift
//  RoutineChart
//
//  Repository protocol for Routine data access
//

import Foundation

protocol RoutineRepository {
    func create(_ routine: Routine) async throws
    func get(id: String) async throws -> Routine?
    func update(_ routine: Routine) async throws
    func getAll(familyId: String, includeDeleted: Bool) async throws -> [Routine]
    func softDelete(id: String) async throws
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/repositories/RoutineRepository.kt
package com.HammersTech.RoutineChart.core.domain.repositories

import com.HammersTech.RoutineChart.core.domain.models.Routine
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for Routine operations
 */
interface RoutineRepository {
    suspend fun create(routine: Routine)

    suspend fun update(routine: Routine)

    suspend fun getById(id: String): Routine?

    fun observeById(id: String): Flow<Routine?>

    suspend fun getAll(
        userId: String,
        familyId: String?,
        includeDeleted: Boolean,
    ): List<Routine>

    fun observeByFamilyId(familyId: String): Flow<List<Routine>> // Keep for backward compatibility, but should use getAll
}
